crates/sargon/src/security_center/client.rs

```diff
@@ -0,0 +1,331 @@
+use crate::prelude::*;
+
+pub struct SecurityCenterClient;
+
+impl SecurityCenterClient {
+    pub fn check_security_problems(
+        input: CheckSecurityProblemsInput,
+    ) -> Vec<SecurityProblem> {
+        let mut problems = Vec::new();
+
+        let is_cloud_profile_sync_enabled =
+            *input.is_cloud_profile_sync_enabled;
+
+        let has_problem_3 = || {
+            if input.unrecoverable_entities.is_empty() {
+                return None;
+            }
+            Some(input.unrecoverable_entities)
+        };
+
+        let has_problem_5 = || {
+            if !is_cloud_profile_sync_enabled {
+                return false;
+            }
+            let Some(cloud_backup) = input.last_cloud_backup else {
+                return true;
+            };
+            *cloud_backup.is_failed
+        };
```

Comment:
The logic for problem 5 is not correct. 🔍 

In Android we define as problem 5 when cloud backup has a service error (e.g. google cloud is down),  which is exactly what is described in [confluence](https://radixdlt.atlassian.net/wiki/spaces/AT/pages/3392569357/Security-related+Problem+States+in+the+Wallet#Security-Problem-5). 

this is how we express the cloud related security problems 
```
sealed interface CloudBackupNotWorking : SecurityProblem {
        // security problem 5
        data class ServiceError(val isAnyActivePersonaAffected: Boolean) : CloudBackupNotWorking
        // security problem 6 & 7
        data class Disabled(
            val isAnyActivePersonaAffected: Boolean,
            val hasManualBackup: Boolean
        ) : CloudBackupNotWorking
    }
```

and this is the logic to define when we have a security problem 5:
```
if (cloudBackupState is BackupState.CloudBackupEnabled && cloudBackupState.hasAnyErrors) {
     SecurityProblem.CloudBackupNotWorking.ServiceError(isAnyActivePersonaAffected = activePersonasNeedCloudBackup > 0)
}
```

cloud backup is enabled (meaning toggle in settings screen is on) but we have errors